src/java.net.http/share/classes/java/net/http/HttpClient.java

```diff
@@ -410,6 +410,12 @@ public interface Builder {
         /**
          * Sets an authenticator to use for HTTP authentication.
          *
+         * @apiNote
+         * If a {@link HttpRequest} has an {@code Authorization} or {@code
+         * Proxy-Authorization} header set then its value will be used and
+         * the {@link Authenticator} will not be invoked for the corresponding
+         * authentication.
+         *
```

Comment:
Reading this makes me wonder if this should be normative, as in part of the spec rather than a note for developers using the API. Has that been discussed?

Reply:
When to use the Authenticator (or not) is typically implementation dependent. We should have documented how our implementation used the authenticator when the API was added in JDK 11. I am not sure we should make this text normative, but if you believe we should I will not object. Do you think we should?

Reply:
Would it be better as an implNote then? I guess it's possible that other implementations of the API might not behave this way.

Reply:
I have changed it to an implNote, referring to the JDK built-in implementation, as elsewhere.